diff --git a/CommonDefs.cmake b/CommonDefs.cmake
index 70d788b..303ec86 100644
--- a/CommonDefs.cmake
+++ b/CommonDefs.cmake
@@ -29,8 +29,8 @@ endforeach(OUTPUTCONFIG CMAKE_CONFIGURATION_TYPES)
 ## Manage static/shared property of external libraries
 
 # Make CMake use static version of all dependencies by default
-option(USE_STATIC_EXT_LIBS "S2OPC libraries and binaries depend on static version of external libraries" ON)
-option(BUILD_SHARED_LIBS "Build dynamic libraries for S2OPC instead of static libraries" OFF)
+option(USE_STATIC_EXT_LIBS "S2OPC libraries and binaries depend on static version of external libraries" OFF)
+option(BUILD_SHARED_LIBS "Build dynamic libraries for S2OPC instead of static libraries" ON)
 if(USE_STATIC_EXT_LIBS)
   set(USE_STATIC_MBEDTLS_LIB ${USE_STATIC_EXT_LIBS})
   set(USE_STATIC_EXPAT_LIB ${USE_STATIC_EXT_LIBS})
diff --git a/src/ClientServer/frontend/client_wrapper/libs2opc_client.h b/src/ClientServer/frontend/client_wrapper/libs2opc_client.h
index 3c187fc..359240f 100644
--- a/src/ClientServer/frontend/client_wrapper/libs2opc_client.h
+++ b/src/ClientServer/frontend/client_wrapper/libs2opc_client.h
@@ -131,7 +131,9 @@ typedef enum
     SOPC_LibSub_DataType_integer = 2,
     SOPC_LibSub_DataType_string = 3,
     SOPC_LibSub_DataType_bytestring = 4,
-    SOPC_LibSub_DataType_other = 5
+    SOPC_LibSub_DataType_float = 5,
+    SOPC_LibSub_DataType_double = 6,
+    SOPC_LibSub_DataType_other = 7
 } SOPC_LibSub_DataType;
 
 /**
diff --git a/src/ClientServer/frontend/client_wrapper/state_machine.c b/src/ClientServer/frontend/client_wrapper/state_machine.c
index b45d556..0d3b595 100644
--- a/src/ClientServer/frontend/client_wrapper/state_machine.c
+++ b/src/ClientServer/frontend/client_wrapper/state_machine.c
@@ -1373,7 +1373,6 @@ static void StaMac_ProcessMsg_PublishResponse(SOPC_StaMac_Machine* pSM, uint32_t
     /* Take note to acknowledge later. There is no ack with KeepAlive. */
     /* TODO: this limits the benefits of having multiple pending PublishRequest, maybe
      * it would be more appropriate to have a list of SeqNumbsToAck... */
-    SOPC_ASSERT(!pSM->bAckSubscr);
     if (0 < pPubResp->NoOfAvailableSequenceNumbers)
     {
         pSM->bAckSubscr = true;
@@ -1402,6 +1401,7 @@ static void StaMac_ProcessMsg_PublishResponse(SOPC_StaMac_Machine* pSM, uint32_t
         {
             pMonItNotif = &pDataNotif->MonitoredItems[i];
             status = Helpers_NewValueFromDataValue(&pMonItNotif->Value, &plsVal);
+            Helpers_Log(SOPC_LOG_LEVEL_INFO, "%s:%d: i=%d, plsVal->type=%d", "StaMac_ProcessMsg_PublishResponse", __LINE__, i, plsVal->type);
             if (SOPC_STATUS_OK == status)
             {
                 if (NULL != pSM->pCbkLibSubDataChanged)
@@ -1637,6 +1637,7 @@ static void StaMac_PostProcessActions(SOPC_StaMac_Machine* pSM, SOPC_StaMac_Stat
     {
     /* Mostly when stActivated is reached */
     case stActivated:
+    case stCreatingMonIt:
         /* add tokens, but wait for at least a monitored item */
         if (0 != pSM->iSubscriptionID && pSM->nTokenUsable < pSM->nTokenTarget)
         {
diff --git a/src/ClientServer/frontend/client_wrapper/toolkit_helpers.c b/src/ClientServer/frontend/client_wrapper/toolkit_helpers.c
index 5d36fe1..8242a65 100644
--- a/src/ClientServer/frontend/client_wrapper/toolkit_helpers.c
+++ b/src/ClientServer/frontend/client_wrapper/toolkit_helpers.c
@@ -120,7 +120,7 @@ SOPC_ReturnStatus Helpers_NewSCConfigFromLibSubCfg(const char* szServerUrl,
                                                            lPathsUntrustedLinks, lPathsIssuedCerts, lPathsCRL, &pPki);
             if (SOPC_STATUS_OK != status)
             {
-                Helpers_Log(SOPC_LOG_LEVEL_ERROR, "Failed to create PKI.");
+                Helpers_Log(SOPC_LOG_LEVEL_ERROR, "Failed to create PKI, status=%d", status);
             }
         }
         else
@@ -602,9 +602,31 @@ SOPC_ReturnStatus Helpers_NewValueFromDataValue(SOPC_DataValue* pVal, SOPC_LibSu
             }
             /* else we leave value NULL and length = 0 */
             break;
-        case SOPC_Null_Id:
         case SOPC_Float_Id:
+            plsVal->type = SOPC_LibSub_DataType_float;
+            plsVal->value = SOPC_Malloc(sizeof(float));
+            if (NULL == plsVal->value)
+            {
+                status = SOPC_STATUS_OUT_OF_MEMORY;
+            }
+            else
+            {
+                *(float*) plsVal->value = (float) pVal->Value.Value.Floatv;
+            }
+            break;
         case SOPC_Double_Id:
+            plsVal->type = SOPC_LibSub_DataType_double;
+            plsVal->value = SOPC_Malloc(sizeof(double));
+            if (NULL == plsVal->value)
+            {
+                status = SOPC_STATUS_OUT_OF_MEMORY;
+            }
+            else
+            {
+                *(double*) plsVal->value = (double) pVal->Value.Value.Doublev;
+            }
+            break;
+        case SOPC_Null_Id:
         case SOPC_DateTime_Id:
         case SOPC_Guid_Id:
         case SOPC_XmlElement_Id:
diff --git a/src/ClientServer/services/b2c/message_out_bs.c b/src/ClientServer/services/b2c/message_out_bs.c
index 5caa845..02b068f 100644
--- a/src/ClientServer/services/b2c/message_out_bs.c
+++ b/src/ClientServer/services/b2c/message_out_bs.c
@@ -258,7 +258,7 @@ static void internal__message_out_bs__encode_msg(const constants__t_channel_conf
             reqHeader = (OpcUa_RequestHeader*) message_out_bs__msg_header;
             reqHeader->Timestamp = SOPC_Time_GetCurrentTimeUTC();
             // TODO: reqHeader->AuditEntryId ?
-            reqHeader->TimeoutHint = SOPC_REQUEST_TIMEOUT_MS / 2; // TODO: to be configured by each service ?
+            reqHeader->TimeoutHint = 0; // TODO: to be configured by each service ?
         }
         else if (&OpcUa_ResponseHeader_EncodeableType == headerType)
         {
diff --git a/src/Common/configuration/sopc_common_constants.h b/src/Common/configuration/sopc_common_constants.h
index cd922b8..ca588e2 100644
--- a/src/Common/configuration/sopc_common_constants.h
+++ b/src/Common/configuration/sopc_common_constants.h
@@ -77,7 +77,7 @@ bool SOPC_Common_SetEncodingConstants(SOPC_Common_EncodingConstants config);
  *  Note: if 0 is chosen SOPC_RECEIVE_MAX_MESSAGE_LENGTH definition shall be changed not to use it and shall not be 0.
  */
 #ifndef SOPC_DEFAULT_RECEIVE_MAX_NB_CHUNKS
-#define SOPC_DEFAULT_RECEIVE_MAX_NB_CHUNKS 5
+#define SOPC_DEFAULT_RECEIVE_MAX_NB_CHUNKS 10
 #endif /* SOPC_DEFAULT_RECEIVE_MAX_NB_CHUNKS */
 
 /** @brief Maximum message length accepted in reception (must be >= SOPC_TCP_UA_MAX_BUFFER_SIZE), 0 means no limit.
